specification/splitio/resource-manager/SplitIO.Experimentation/preview/2024-03-01-preview/splitio.json

```diff
@@ -0,0 +1,797 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "workspaceName",
+            "in": "path",
+            "description": "The name of the ExperimentationWorkspace resource.",
+            "required": true,
+            "type": "string",
+            "minLength": 5,
+            "maxLength": 50,
+            "pattern": "^[a-zA-Z0-9](?!.*--)[a-zA-Z0-9-]*[a-zA-Z0-9]$"
+          },
+          {
+            "name": "applicationId",
+            "in": "path",
+            "description": "The id of the Split.IO data plane authentication application.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "exchangeTokenParameters",
+            "in": "body",
+            "description": "The parameter of exchange token action.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ExchangeTokenParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ExchangeTokenResult"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DataPlaneApplicationsExchangeToken": {
+            "$ref": "./examples/DataPlaneApplicationsExchangeToken.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AccessPolicy": {
+      "type": "object",
+      "description": "Access policy for an experimentation workspace.",
+      "properties": {
+        "applicationId": {
+          "type": "string",
+          "description": "Application id used for access to the experimentation workspace."
+        },
+        "dataPlaneEndpoint": {
+          "type": "string",
+          "description": "Endpoint used for access to the experimentation workspace.",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "applicationId"
+      ]
+    },
+    "AccessPolicyUpdate": {
+      "type": "object",
+      "description": "Access policy for an experimentation workspace.",
+      "properties": {
+        "applicationId": {
+          "type": "string",
+          "description": "Application id used for access to the experimentation workspace."
+        }
+      }
+    },
+    "Azure.ResourceManager.ResourceProvisioningState": {
+      "type": "string",
+      "description": "The provisioning state of a resource type.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled"
+      ],
```

Comment:
This provisioningState is returned when the PUT/PATCH call is made too. 

Don't we need a starting/intermediate state in those responses , before eventually responding with the terminal state.

Reply:
Updated to include more states.